### PR TITLE
User dashboard

### PR DESCRIPTION
**Pull request by lauradurandv**
I added search functionality and the the ability to edit user profiles.
Captura de pantalla 2024-08-31 a la(s) 4 58 14 p m
Captura de pantalla 2024-08-31 a la(s) 4 58 01 p m

### CODE REVIEW

File: server/controllers/userController.mjs [hunk 1]
```diff
@@ -207,40 +207,50 @@ export const resetPassword = async (req, res) => {
...
-  // Prepare the updates object
   const updates = {};
+
   if (name) {
     if (typeof name !== 'string' || name.trim() === '') {
       return res.status(400).json({ error: 'Invalid name' });
     }
-    updates.name = name; // Update name field in Firebase Authentication
+    updates.displayName = name;
+    updates.name = name; // For Firestore
```
Comment (SonaliBedge) [inline server/controllers/userController.mjs:223]: We don't have a displayName field in the userModel.mjs file. Here are the fields for the user model:
 " name,
    email,
    userType,
    profilePictureUrl = null, // Default image URL
    createdAt = new Date(),"

File: server/controllers/userController.mjs [hunk 2]
```diff
@@ -207,40 +207,50 @@ export const resetPassword = async (req, res) => {
...
   if (profilePictureUrl) {
     if (
       typeof profilePictureUrl !== 'string' ||
       !profilePictureUrl.startsWith('http')
     ) {
       return res.status(400).json({ error: 'Invalid profilePictureUrl' });
     }
-    updates.photoURL = profilePictureUrl; // Update photoURL field in Firebase Authentication
+    updates.photoURL = profilePictureUrl;
+    updates.profilePictureUrl = profilePictureUrl; // For Firestore
```
Comment (SonaliBedge) [inline server/controllers/userController.mjs:241]: same here

File: server/models/UsersModel.mjs [hunk 3]
```diff
@@ -35,6 +35,16 @@ class UsersModel {
...
+
+  static async update(uid, updates) {
+    const db = admin.firestore();
+    try {
+      await db.collection('users').doc(uid).update(updates);
+    } catch (error) {
+      console.error('Error updating user:', error);
+      throw error;
+    }
+  }
```
Comment (SonaliBedge) [inline server/models/UsersModel.mjs:47]: I think this update() should be in userController.mjs file not in UsersModel.mjs file.

File: src/services/api.js [hunk 4]
```diff
@@ -109,11 +109,35 @@ export const AddUserCourse = async (userId, courseId , role) => {
...
+      },
+      {
+        headers: {
+          Authorization: `Bearer ${token}`,
+        },
+      },
+    );
+    return response.data; // Return the updated profile data
+  } catch (error) {
+    console.error('Error updating profile:', error);
```
Comment (SonaliBedge) [inline src/services/api.js:140]: Nice work!

File: server/controllers/userController.mjs (hunk 1 above)
Comment (EdithHarrison) [reply server/controllers/userController.mjs]: addressed!

File: server/controllers/userController.mjs (hunk 2 above)
Comment (EdithHarrison) [reply server/controllers/userController.mjs]: addressed!

File: server/models/UsersModel.mjs (hunk 3 above)
Comment (EdithHarrison) [reply server/models/UsersModel.mjs]: addressed!

**Review by SonaliBedge — APPROVED** [review]

**Review by EdithHarrison — APPROVED** [review]

**Review by SonaliBedge — APPROVED** [review]

**Pull request merged**